ledger/committee/src/lib.rs

```diff
@@ -0,0 +1,352 @@
+        Ok(Self { round, members, total_stake })
+    }
+
+    /// Returns a new `Committee` instance for the next round.
+    /// TODO (howardwu): Add arguments for members (and stake) 1) to be added, 2) to be updated, and 3) to be removed.
+    pub fn to_next_round(&self) -> Self {
+        // Return the new committee.
+        Self { round: self.round.saturating_add(1), members: self.members.clone(), total_stake: self.total_stake }
+    }
+}
+
+impl<N: Network> Committee<N> {
+    /// Returns the current round number.
+    pub const fn round(&self) -> u64 {
+        self.round
+    }
+
+    /// Returns the committee members alongside their stake.
+    pub const fn members(&self) -> &IndexMap<Address<N>, u64> {
+        &self.members
+    }
+
+    /// Returns the number of validators in the committee.
+    pub fn num_members(&self) -> usize {
+        self.members.len()
+    }
+
+    /// Returns `true` if the given address is in the committee.
+    pub fn is_committee_member(&self, address: Address<N>) -> bool {
+        self.members.contains_key(&address)
+    }
+
+    /// Returns `true` if the combined stake for the given addresses reaches the availability threshold.
+    /// This method takes in a `HashSet` to guarantee that the given addresses are unique.
+    pub fn is_availability_threshold_reached(&self, addresses: &HashSet<Address<N>>) -> bool {
+        let mut stake = 0u64;
+        // Compute the combined stake for the given addresses.
+        for address in addresses {
+            // Accumulate the stake, checking for overflow.
+            stake = stake.saturating_add(self.get_stake(*address));
+        }
+        // Return whether the combined stake reaches the availability threshold.
+        stake >= self.availability_threshold()
+    }
+
+    /// Returns `true` if the combined stake for the given addresses reaches the quorum threshold.
+    /// This method takes in a `HashSet` to guarantee that the given addresses are unique.
+    pub fn is_quorum_threshold_reached(&self, addresses: &HashSet<Address<N>>) -> bool {
+        let mut stake = 0u64;
+        // Compute the combined stake for the given addresses.
+        for address in addresses {
+            // Accumulate the stake, checking for overflow.
+            stake = stake.saturating_add(self.get_stake(*address));
+        }
+        // Return whether the combined stake reaches the quorum threshold.
+        stake >= self.quorum_threshold()
+    }
+
+    /// Returns the amount of stake for the given address.
+    pub fn get_stake(&self, address: Address<N>) -> u64 {
+        self.members.get(&address).copied().unwrap_or_default()
+    }
+
+    /// Returns the amount of stake required to reach the availability threshold `(f + 1)`.
+    pub fn availability_threshold(&self) -> u64 {
+        // Assuming `N = 3f + 1 + k`, where `0 <= k < 3`,
+        // then `(N + 2) / 3 = f + 1 + k/3 = f + 1`.
+        self.total_stake().saturating_add(2) / 3
```

Comment:
nit: Use `saturating_div` instead of `/`

```suggestion
        self.total_stake().saturating_add(2).saturating_div(3)
```